src/cryptonote_protocol/cryptonote_protocol_handler.h

```diff
@@ -112,6 +112,15 @@ namespace cryptonote
     void log_connections();
     std::list<connection_info> get_connections();
     const block_queue &get_block_queue() const { return m_block_queue; }
+    const std::uint64_t max_average_of_blocksize_in_queue() {
+      std::vector<std::uint64_t> average_blocksize{0};
```

Comment:
This initializes with one element? The overloads for this are annoying.

Reply:
Thanks. Will be fixed in updated PR.

Reply:
Regarding I just remembered one thing. I intentionally create a vector with single `0` element in case that `m_block_queue` is empty. In that specific case I want that the method to return `0` in line 121.